tests/bulk_create/tests.py

```diff
@@ -293,6 +295,22 @@ def test_explicit_batch_size_respects_max_batch_size(self):
         with self.assertNumQueries(ceil(len(objs) / max_batch_size)):
             Country.objects.bulk_create(objs, batch_size=max_batch_size + 1)
 
+    @skipUnlessDBFeature("has_bulk_insert")
+    @skipUnless(connection.vendor == "sqlite", "SQLite-specific test.")
+    def test_max_batch_size_respects_sqlite_variable_limit(self):
+        objs = [Country(name=f"Country {i}") for i in range(1000)]
+        fields = ["name", "iso_two_letter", "description"]
+        limit_name = sqlite3.SQLITE_LIMIT_VARIABLE_NUMBER
+        current_limit = connection.features.max_query_params
+        new_limit = min(640, current_limit)
+        try:
+            connection.connection.setlimit(limit_name, new_limit)
+            max_batch_size = new_limit // len(fields)
+            with self.assertNumQueries(ceil(len(objs) / max_batch_size)):
+                Country.objects.bulk_create(objs)
+        finally:
+            connection.connection.setlimit(limit_name, current_limit)
+
```

Comment:
I appreciate the test but I'm not sure the issue warrants testing `bulk_batch_size` through SQLite specifics.

IMO making sure that `bulk_batch_size` works properly (like you did in `test_max_query_params_respects_variable_limit` and `test_bulk_batch_size_respects_variable_limit`) and then keeping `bulk_create` and `bulk_update` test focused on the `bulk_batch_size` makes more sense.

In other words, we're not testing the Oracle implementation of `bulk_batch_size` here so why should we do it for SQLite?

Reply:
Good point, I felt the same way too. I wrote these ones mostly to confirm it for myself and for completeness sake, but I wasn't too happy with having vendor-specific tests in there either.

I've added a commit to remove them. Will squash or let the mergers squash them when we're happy with the overall patch. Thanks!